bin/reth-bench/README.md

```diff
@@ -31,35 +31,100 @@ Otherwise, running `make maxperf` at the root of the repo should be sufficient f
-    ```
+Starting with a synced ethereum mainnet node, to run a benchmark starting at block 21,000,000, the node would need to be unwound first:
+```bash
+reth stage unwind to-block 21000000
+```
 
-    Replace `<start_block>`, `<end_block>`, and `<jwt_file_path>` with the appropriate values for your testing environment. `<rpc-url>` should be the URL of an RPC endpoint that can provide the blocks that will be used during the execution.
-    Note that this assumes that the benchmark node's engine API is running on `http://127.0.0.1:8551`, which is set as a default value in `reth-bench`. To configure this value, use the `--engine-rpc-url` flag.
+The following `reth-bench` command would then start the benchmark at block 21,000,000:
+```bash
+reth-bench new-payload-fcu --rpc-url <rpc-url> --from 21000000 --to <end_block> --jwtsecret <jwt_file_path>
+```
 
- 3. **Observe Outputs**: Upon running the command, `reth-bench` will output benchmark results, showing processing speeds and gas usage, which are crucial for analyzing the node's performance.
+Finally, make sure that reth is built using a build profile suitable for what you are trying to measure.
+For example, if the purpose of the benchmark is to load test and measure `reth`'s maximum speed, it would be compiled with the `maxperf` profile:
+```bash
+make maxperf
+```
 
-    Example output:
-    ```
-    2024-05-30T00:45:20.806691Z  INFO Running benchmark using data from RPC URL: http://<rpc-url>:8545
-    // ... logs per block
-    2024-05-30T00:45:34.203172Z  INFO Total Ggas/s: 0.15 total_duration=5.085704882s total_gas_used=741620668.0
-    ```
+If the purpose of the benchmark is for analyzing flamegraphs for a specific routine using `perf` or `samply`, it should be compiled with the `profiling` profile:
+```bash
+make profiling
+```
 
- 4. **Stop and Review**: Once the benchmark completes, terminate the `reth` process and review the logs and performance metrics collected, if any.
- 5. **Repeat**.
+If the purpose of the benchmark is to obtain `jemalloc` memory profiles that can then be analyzed by `jeprof`, it should be compiled with the `profiling` profile and the `jemalloc-prof` feature:
+```bash
+RUSTFLAGS="-C target-cpu=native" cargo build --profile profiling --features "jemalloc-prof,asm-keccak"
+```
+
+> [!NOTE]
+> Jemalloc memory profiling currently only works on linux.
+
+Finally, if the purpose of the benchmark is to profile the node when `snmalloc` is configured as the default allocator, it would be built with the following
+command:
+```bash
+RUSTFLAGS="-C target-cpu=native" cargo build --profile profiling --no-default-features --features "snmalloc-native,asm-keccak"
+```
+
+### Run the Benchmark:
+First, start the reth node. Here is an example that runs `reth` compiled with the `profiling` profile, runs `samply`, and configures `reth` to run with metrics enabled:
+```bash
+samply record -p 3001 target/profiling/reth node --metrics localhost:9001
```

Comment:
nit: given that we refer later to jwt secret maybe good to specify it here too, so that we make it clear that it should be the same file
```suggestion
samply record -p 3001 target/profiling/reth node --metrics localhost:9001 --authrpc.jwtsecret <jwt_file_path>
```